internal/pkg/levee/propagation/propagation.go

```diff
@@ -0,0 +1,350 @@
+package propagation
+
+import (
+	"fmt"
+	"go/types"
+	"log"
+	"strings"
+
+	"github.com/google/go-flow-levee/internal/pkg/fieldtags"
+	"github.com/google/go-flow-levee/internal/pkg/sanitizer"
+	"github.com/google/go-flow-levee/internal/pkg/source"
+	"github.com/google/go-flow-levee/internal/pkg/utils"
+	"golang.org/x/tools/go/pointer"
+	"golang.org/x/tools/go/ssa"
+)
+
+type DFSRecord struct {
+	Root         ssa.Node
+	Marked       map[ssa.Node]bool
+	PreOrder     []ssa.Node
+	Sanitizers   []*sanitizer.Sanitizer
+	Config       source.Classifier
+	TaggedFields fieldtags.ResultType
+}
+
+func Dfs(n ssa.Node, conf source.Classifier, taggedFields fieldtags.ResultType) DFSRecord {
+	record := DFSRecord{
+		Root:         n,
+		Marked:       make(map[ssa.Node]bool),
+		PreOrder:     nil,
+		Sanitizers:   nil,
+		Config:       conf,
+		TaggedFields: taggedFields,
+	}
+	var lastBlockVisited *ssa.BasicBlock = nil
```

Comment:
`= nil` is not needed.

Reply:
In fact, consider inlining this variable.

Reply:
I had been under the impression that the dfs would mutate that value, but that appears to be incorrect.  `nil` passed directly to consumers.